File: app/src/main/java/com/alexvasilkov/events/cache/CacheProvider.java
package com.alexvasilkov.events.cache;

import com.alexvasilkov.events.Event;
import com.alexvasilkov.events.EventResult;

import androidx.annotation.NonNull;

public interface CacheProvider {

    EventResult loadFromCache(@NonNull Event event) throws Exception;

    void saveToCache(@NonNull Event event, EventResult result) throws Exception;
}

File: app/src/main/java/com/alexvasilkov/events/cache/CacheProvider.kt
package com.alexvasilkov.events.cache

import com.alexvasilkov.events.Event
import com.alexvasilkov.events.EventResult

interface CacheProvider {

    @Throws(Exception::class)
    fun loadFromCache(event: Event): EventResult

    @Throws(Exception::class)
    fun saveToCache(event: Event, result: EventResult)
}